python/cuml/tests/test_linear_model.py

```diff
@@ -193,10 +204,36 @@ def test_linear_regression_single_column():
         model.fit(cp.random.rand(46341), cp.random.rand(46341))
 
 
-@pytest.mark.parametrize("datatype", [np.float32, np.float64])
-def test_linear_regression_model_default(datatype):
-
-    X_train, X_test, y_train, y_test = small_regression_dataset(datatype)
+@given(
+    split_datasets(
+        regression_datasets(
+            # Two assumptions required for cuml.LinearRegression:
+            n_samples=st.integers(
+                min_value=20, max_value=200
```

Comment:
We could have a different number of examples in PR runs vs nightly runs, potentially with the existing unit vs stress test mechanisms, what do you think?

Reply:
Yes, @wphicks and I had already discussed something along those lines. We should also consider to simply change the hypothesis defaults globally instead of adjusting them test-by-test.

As a note: The referenced line has no influence on the number of tested examples, it only controls the range of possible dataset sizes.

Reply:
Implemented via [hypothesis profiles](https://hypothesis.readthedocs.io/en/latest/settings.html#settings-profiles) in https://github.com/rapidsai/cuml/pull/4952/commits/fab7c725eeaec985ea4bf2edff7bcf22e5a0f0bc .